### ISSUE TITLE
Accessibility: Reset password form missing required indicator

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43660]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The reset password form on the My Account page does not have a required indicator.
<img width="1044" alt="Screenshot of reset password form" src="https://github.com/woocommerce/woocommerce/assets/13153456/ef2f247a-db39-4a22-8e0e-5e859f364403">

Relevant code: https://github.com/woocommerce/woocommerce/blob/33ae0d607df0704e59025bb1b33cdcb96b497b0f/plugins/woocommerce/templates/myaccount/form-lost-password.php#L28

### Expected behavior

All required fields should have a visible and auditory required indicator.

Add an aria-label to the asterisk or change it to visible required text (preferred).
A) Best:
`<label for="user_login"><?php esc_html_e( 'Username or email', 'woocommerce' ); ?></label>&nbsp;<span class="required">(required)</span></label>`

B) Sufficient:
`<label for="user_login"><?php esc_html_e( 'Username or email', 'woocommerce' ); ?></label>&nbsp;<span class="required" aria-label="required">*</span></label>`

Add the required attribute to the inputs like this:
<input class="woocommerce-Input woocommerce-Input--text input-text" type="text" name="user_login" id="user_login" autocomplete="username" required />

### Actual behavior

No required indicator for sighted or screen reader users.

### Steps to reproduce

See code linked above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
